256 add templates for mams integration points

Anoopsinghrawat opened the pull request.

kwathen reviewed (changes requested):
It looks like each analysis template has 4 options for returns.   Can you please add a comment to the code so that user is clear that only one of the return blocks should be used and please make it clear that the user should remove the return blocks that are not needed for their specific case. 

If there is a preference, for example, option 1 is preferred because ...
If the other options provided add value, what and why would someone use the other options?

Anoopsinghrawat commented:
Added comments in the code to ensure user deletes the return statements that are not required. Also added some comments to further highlight the differences between the options and when one would be preferred over the others.
